hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryFunction.java

```diff
@@ -106,7 +106,7 @@ public HazelcastJsonValue eval(final Row row, final ExpressionEvalContext contex
 
         final JsonPath jsonPath;
         try {
-            jsonPath = pathCache.asMap().computeIfAbsent(path, JsonPathUtil::compile);
+            jsonPath = pathCache.computeIfAbsent(path, JsonPathUtil::compile);
```

Comment:
We can cache compiled expressions the same way as in `JsonValueFunction`

Reply:
Done